src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -2641,16 +2771,198 @@ public override object PayloadValue(int index)
+        /// The number of buckets.
+        /// </summary>
+        public int Count { get { return GetInt16At(12); } }
+
+        public GCFitBucket Buckets(int bucketIndex)
+        {
+            Debug.Assert(0 <= bucketIndex && bucketIndex < Count);
+            int baseAddress = 14 + SizeOfGCFitBucket * bucketIndex;
+            int index = GetInt32At(baseAddress);
+            int count = GetInt32At(baseAddress + 4);
+            long size = (PointerSize == 8) ? GetInt64At(baseAddress + 8) : (long)GetInt32At(baseAddress + 8);
+            return new GCFitBucket(index, count, size);
+        }
+
+
+        #region Private
+        internal GCFitBucketInfoTraceData(Action<GCFitBucketInfoTraceData> action, int eventID, int task, string taskName, Guid taskGuid, int opcode, string opcodeName, Guid providerGuid, string providerName)
+            : base(eventID, task, taskName, taskGuid, opcode, opcodeName, providerGuid, providerName)
+        {
+            Action = action;
+        }
+        protected internal override void Dispatch()
+        {
+            Action(this);
+        }
+        protected internal override Delegate Target
+        {
+            get { return Action; }
+            set { Action = (Action<GCFitBucketInfoTraceData>)value; }
+        }
+        protected internal override void Validate()
+        {
+            int size0 = 14 + Count * SizeOfGCFitBucket;
+            Debug.Assert(!(Version == 0 && EventDataLength != size0));
+            Debug.Assert(!(Version > 0 && EventDataLength < size0));
+        }
+        public override StringBuilder ToXml(StringBuilder sb)
+        {
+            Prefix(sb);
+            XmlAttrib(sb, "ClrInstanceID", ClrInstanceID);
+            XmlAttrib(sb, "BucketKind", BucketKind);
+            XmlAttrib(sb, "TotalSize", TotalSize);
+            XmlAttrib(sb, "Count", Count);
+            sb.Append("/>");
+            return sb;
+        }
+
+        public override string[] PayloadNames
+        {
+            get
+            {
+                if (payloadNames == null)
+                {
+                    payloadNames = new string[] { "ClrInstanceID", "BucketKind", "TotalSize", "Count" };
+                }
+
+                return payloadNames;
+            }
+        }
+
+        public override object PayloadValue(int index)
+        {
+            switch (index)
+            {
+                case 0:
                     return ClrInstanceID;
+                case 1:
+                    return BucketKind;
+                case 2:
+                    return TotalSize;
+                case 3:
+                    return Count;
+
                 default:
                     Debug.Assert(false, "Bad field index");
                     return null;
             }
         }
 
-        private event Action<GenAwareEndTraceData> m_target;
+        private int SizeOfGCFitBucket
+        {
+            get
+            {
+                return PointerSize == 8 ? 16 : 12;
```

Comment:
Same here.  Can you please add a comment that describes the field order and size?